src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -310,7 +310,11 @@ private async Task<bool> InstallPackageAsync(RemoteMatch installItem, NuGetv3Loc
                         await _logger.LogAsync(e.AsLogMessage());
                     }
 
-                    await _logger.LogMessagesAsync(e.Results.SelectMany(p => p.Issues));
+                    if (e.Results != null)
```

Comment:
Why change this? When is this null? 

Seems like a good change, bu I want to understand the scenario where you found it.

Reply:
If the package is unsigned and unsigned packages are not allowed a `SignatureException` will be thrown but it won't have results because it didn't went through any Verification provider. Therefore we need to be aware of it.

Reply:
Might be worth adding a comment to this effect, because the reason isn't obvious.